i6_models/config.py

```diff
@@ -51,3 +53,22 @@ def _validate_types(self) -> None:
 
     def __post_init__(self) -> None:
         self._validate_types()
+
+
+ConfigType = TypeVar("ConfigType", bound=ModelConfiguration)
+ModuleType = TypeVar("ModuleType", bound=nn.Module)
+
+
+@dataclass
+class SubassemblyFactory:
+    """
+    Dataclass for a combination of a Subassembly/Part and the corresponding configuration.
+    Also provides a function to construct the corresponding object through this dataclass
+    """
+
+    module_class: Type[ModuleType]
```

Comment:
```suggestion
    module_class: Callable[[ConfigType], ModuleType]
```

Reply:
I am not sure about this. I thought `Type` was meant for class objects.

Reply:
Yes but each class is also a callable due to its constructor. This typehint ensures that the constructor takes a single parameter of type `ConfigType` so that calling `self.module_class(self.cfg)` is valid.